baseapp/setters.go

```diff
@@ -74,6 +74,12 @@ func (app *BaseApp) Router() Router {
 	}
 	return app.router
 }
+func (app *BaseApp) QueryRouter() QueryRouter {
```

Comment:
What does sealing do on the baseapp and do we need to seal the query router as well?  anything else we're missing?  docs would be helpful!

Reply:
I'm not sure why we have the seal it.  I just saw that the router was sealed, so I sealed the QueryRouter as well

Reply:
Hmm, I think we should restrict sealing to fields that actually need to be sealed for clarity, otherwise readers of the code will have a hard time figuring out what our intent was (as apparently we ourselves have a hard time figuring out already!). As I understand, this is just defensive coding - I believe @mossid wrote the original PR - and we only need to do it for write-once fields or fields that should not be read from after a certain event in the app lifecycle.

Reply:
Also, don't we *want* to query the QueryRouter after the app is sealed?